pyproject.toml

```diff
@@ -153,26 +153,23 @@ rapids = ["cudf>=0.9", "cuml>=0.9", "cugraph>=0.9"]  # GPU accelerated calculati
 dask = ["dask[array]>=2022.09.2"]  # Use the Dask parallelization engine
 dask-ml = ["dask-ml", "scanpy[dask]"]  # Dask-ML for sklearn-like API
 
-[tool.hatch.build]
-exclude = ["scanpy/tests"]
 [tool.hatch.build.targets.wheel]
-# TODO: move scanpy to src/
-only-include = ["src/testing", "scanpy"]
-sources = ["src", "."]
+packages = ["src/testing", "src/scanpy"]
 [tool.hatch.version]
 source = "vcs"
 [tool.hatch.build.hooks.vcs]
-version-file = "scanpy/_version.py"
+version-file = "src/scanpy/_version.py"
```

Comment:
Curiosity of mine: This file does not exist, right? What is it referring to?

Reply:
the line you’re commenting on causes the file to be created by `hatch-vcs`:

https://github.com/scverse/scanpy/blob/ef6af20ab3dfcd784f9edd9ea8dc426835518e18/pyproject.toml#L3

That happens whenever the project is built, e.g. when doing `python -m build` to build a wheel or when doing `pip install -e .` to install the project in editable mode.

Reply:
Personally I think `__version__` is silly and shouldn’t exist, and neither should that file. Package metadata is all that should matter.